File: tuts/003-s3-gettingstarted/dotnet/S3Wrapper.cs
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

using Amazon.S3;
using Microsoft.Extensions.Logging;

public class S3Wrapper
{
    private readonly IAmazonS3 _client;
    private readonly ILogger<S3Wrapper> _logger;

    public S3Wrapper(IAmazonS3 client, ILogger<S3Wrapper> logger)
    {
        _client = client;
        _logger = logger;
    }

    // TODO: Add async wrapper methods matching CLI tutorial actions
}

File: tuts/003-s3-gettingstarted/ruby/s3_wrapper.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

require 'aws-sdk-s3'
require 'logger'

class S3Wrapper
  def initialize(client, logger: Logger.new($stdout))
    @client = client
    @logger = logger
  end

  # TODO: Add wrapper methods matching CLI tutorial actions
end
